### PR TITLE
add support for running 'pip check' during sanity check in generic PythonPackage easyblock

### PR DESCRIPTION
**Pull request by boegel**
cc @Flamefire

### CODE REVIEW

File: easybuild/easyblocks/generic/pythonpackage.py [hunk 1]
```diff
@@ -645,6 +660,16 @@ def sanity_check_step(self, *args, **kwargs):
                 exts_filter = (orig_exts_filter[0].replace('python', self.python_cmd), orig_exts_filter[1])
                 kwargs.update({'exts_filter': exts_filter})
 
+        if self.cfg.get('sanity_pip_check', False):
+            pip_version = det_pip_version()
+            if pip_version:
+                if LooseVersion(pip_version) >= LooseVersion(pip_version):
```
Comment (wpoely86) [inline easybuild/easyblocks/generic/pythonpackage.py]: say what?

Comment (boegel) [reply easybuild/easyblocks/generic/pythonpackage.py]: Ugh. Nice catch.

Comment (boegel) [reply easybuild/easyblocks/generic/pythonpackage.py]: fixed in ea82401

File: easybuild/easyblocks/generic/pythonpackage.py [hunk 2]
```diff
@@ -177,6 +177,24 @@ def det_pylibdir(plat_specific=False, python_cmd=None):
     return pylibdir
 
 
+def det_pip_version():
```
Comment (Flamefire) [inline easybuild/easyblocks/generic/pythonpackage.py]: I'd refactor this to only do the required work: 
```suggestion
def det_pip_version():
 """Determine version of currently active 'pip' command."""

    out, _ = run_cmd("pip --version", verbose=False, simple=False, trace=False)

    pip_version_regex = re.compile('^pip ([0-9.]+)')
    res = pip_version_regex.search(out)
    if res:
        return res.group(1)
    else:
        log = fancylogger.getLogger('det_pylibdir', fname=False)
        log.warning("Failed to determine pip version from '%s' using pattern '%s'", out, pip_version_regex.pattern)
        return None
```

File: easybuild/easyblocks/generic/pythonpackage.py [hunk 3]
```diff
@@ -645,6 +660,16 @@ def sanity_check_step(self, *args, **kwargs):
                 exts_filter = (orig_exts_filter[0].replace('python', self.python_cmd), orig_exts_filter[1])
                 kwargs.update({'exts_filter': exts_filter})
 
+        if self.cfg.get('sanity_pip_check', False):
+            pip_version = det_pip_version()
+            if pip_version:
+                if LooseVersion(pip_version) >= LooseVersion('9.0.0'):
+                    run_cmd("pip check")
```
Comment (wpoely86) [inline easybuild/easyblocks/generic/pythonpackage.py]: this is non - fatal if it fails?

Comment (boegel) [reply easybuild/easyblocks/generic/pythonpackage.py]: It's fatal, `run_cmd` will raise an error for commands that exit with a non-zero exit code by default.

File: easybuild/easyblocks/generic/pythonpackage.py (hunk 2 above)
Comment (boegel) [reply easybuild/easyblocks/generic/pythonpackage.py]: I'll enhance the function a bit to use more logging instead...

**Pull request merged**